src/main/kotlin/javabot/dao/ChatGPTDao.kt

```diff
@@ -0,0 +1,42 @@
+package javabot.dao
+
+import com.google.inject.Inject
+import gpt.api.GPT
+import gpt.models.Message
+import gpt.models.MessageModel
+import io.dropwizard.util.Duration
+import javabot.JavabotConfig
+import javabot.operations.throttle.BotRateLimiter
+
+
+class ChatGPTDao
+@Inject
+constructor(val javabotConfig: JavabotConfig) {
+    companion object {
+        private var limiter: BotRateLimiter? = null
+
+        fun acquire(javabotConfig: JavabotConfig): Boolean {
+            if (limiter == null) {
+                limiter = BotRateLimiter(
+                    javabotConfig.chatGptLimit(),
+                    Duration.days(1).toMilliseconds()
+                )
+            }
+            return limiter!!.tryAcquire()
+        }
+    }
+
+    fun sendPromptToChatGPT(prompt: String): String? {
+        return if (javabotConfig.chatGptKey().isNotEmpty() && acquire(javabotConfig)) {
```

Comment:
I would add the key check to the limiter.  no key?  no acquire. over the limit?  no acquire.  maybe make the limiter a singleton injection at that point.